### PR TITLE
Added data-types.md in the kotlin folder

### LINKED ISSUE #1376: [Concept Entry] Kotlin: Data-Types
### Reason/inspiration (optional)

We would like a new concept entry for [Kotlin](https://www.codecademy.com/resources/docs/kotlin) about data types.

The entry should go in a new file in docs/content/kotlin/concepts/data-types/data-types.md

The entry should

- List all the basic data types and what they hold.
- Show examples of declaring a variable with a particular data type.

Look at the Data Type entries for other languages to see what we'd like.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1376

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
I approved the prior PR with this for a second review. So this has effectively already been reviewed once.

**Review by yangc95 — COMMENTED** [review]
@dakshdeepHERE I left some comments/suggestions for this entry! When you're done looking over everything please tag me.

I may have some more suggestions or direct edits on my end, prior to merge. Thanks for expanding our Kotlin documentation!

File: content/kotlin/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+## Non-Primitive Data Types
+
+The following are non-primitive data types in Kotlin :
+
+- Arrays
+- String
+- Any
+- Nothing
+- Unit
```
[the remark this answers is not in the document]
Comment (dakshdeepHERE) [reply content/kotlin/concepts/data-types/data-types.md:267]: Alright mam I will always keep this in mind for future commits !

File: content/kotlin/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+### Arrays
+
+In Kotlin, an `Array` is a data structure that stores a fixed-size collection of elements of the same data type. Arrays are non-primitive data types, which means they are derived from primitive data types or other non-primitive data types.
+
+To declare an `Array` in Kotlin, you can use the `arrayOf()` function and specify the elements of the array within the parentheses. For example:
```
[the remark this answers is not in the document]
Comment (dakshdeepHERE) [reply content/kotlin/concepts/data-types/data-types.md:278]: Alright wait will change this sentence

File: content/kotlin/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+- Nothing
+- Set
+- String
+- Unit
+
+### Arrays
+
+In Kotlin, an `Array` is a data structure that stores a fixed-size collection of elements of the same data type. Arrays are non-primitive data types, which means they are derived from primitive data types or other non-primitive data types.
+
+To declare an Array in Kotlin, the `arrayOf()` function can be utilized, with the elements of the array specified within the parentheses. For instance:
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/data-types/data-types.md:278]: @yangc95 I removed "you" and changed the sentence like this check if this is right?

**Comment by dakshdeepHERE** [comment]
@yangc95 I have done the changes that you requested and please check the line 278 that you wanted me change if its right or I need to further modify it.

**Comment by yangc95** [comment]
@dakshdeepHERE Thanks for your patience with this PR. I'm going to make some direct changes to this entry and merge after.

cc: @SSwiniarski

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@dakshdeepHERE live link: https://www.codecademy.com/resources/docs/kotlin/data-types

Screen Shot 2023-01-16 at 8 16 30 PM
You're all set!

cc: @SSwiniarski

**Comment by dakshdeepHERE** [comment]
Thank you @yangc95

**Pull request merged**